openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -100,30 +104,93 @@ public OpenAPIConverterService() {
     }
 
     /**
-     * This API is used to return the List of {@code OASResult} that containing all the service node generated yaml
-     * content, json content, openapi content and all diagnostics and error message if generation failed.
+     * This API is used to return the Json Array of {@code OASResult} that containing all the service node generated
+     * yaml content, json content, openapi content and all diagnostics and error message if generation failed.
      */
     @JsonRequest
     public CompletableFuture<OpenAPIConverterResponse> generateOpenAPI(OpenAPIConverterRequest request) {
         return CompletableFuture.supplyAsync(() -> {
             OpenAPIConverterResponse response = new OpenAPIConverterResponse();
-            String fileUri = request.getDocumentFilePath();
-            Optional<SyntaxTree> syntaxTree = getPathFromURI(fileUri).flatMap(workspaceManager::syntaxTree);
-            Optional<SemanticModel> semanticModel = getPathFromURI(fileUri).flatMap(workspaceManager::semanticModel);
-            if (semanticModel.isEmpty() || syntaxTree.isEmpty()) {
-                StringBuilder errorString = getErrorMessage(syntaxTree, semanticModel);
-                response.setError(errorString.toString());
-            } else {
-                response.setError(null);
-                List<OASResult> oasResult = ServiceToOpenAPIConverterUtils.generateOAS3Definition(syntaxTree.get(),
-                        semanticModel.get(), null, false, null);
-                //Response handle with returning list of {@code OASResult} model.
-                response.setContent(oasResult);
+            Path filePath = Path.of(request.getDocumentFilePath());
+            Optional<SemanticModel> semanticModel = workspaceManager.semanticModel(filePath);
+            Optional<Module> module = workspaceManager.module(filePath);
+
+            if (semanticModel.isEmpty()) {
+                response.setError("Error while generating semantic model.");
+                return response;
+            }
+            if (module.isEmpty()) {
+                response.setError("Error while getting the module.");
+                return response;
             }
+            Module defaultModule = module.get();
+            JsonArray specs = new JsonArray();
+            for (DocumentId currentDocumentID : defaultModule.documentIds()) {
+                Document document = defaultModule.document(currentDocumentID);
+
+                SyntaxTree syntaxTree = document.syntaxTree();
```

Comment:
Why can't we directly use ` workspaceManager.syntaxTree()` to get syntax tree instead of going through the module :) ?